Unable to use the replay mode due to unknown substitution: arg

**Bug description**
I've installed ouster-ros package within a ROS2 Humble conda environment onto an Ubuntu 22.04 machine.  
I'm currently able to use the [sensor mode](https://github.com/ouster-lidar/ouster-ros?tab=readme-ov-file#sensor-mode) and the [recording mode](https://github.com/ouster-lidar/ouster-ros?tab=readme-ov-file#recording-mode); unfortunately when I try to use the [replay mode](https://github.com/ouster-lidar/ouster-ros?tab=readme-ov-file#replay-mode) I get the following error:  

```
[INFO] [launch]: All log files can be found below /home/gcervettini/.ros/log/2024-09-16-17-37-20-503076-30559
[INFO] [launch]: Default logging verbosity is set to INFO
[ERROR] [launch]: Caught exception in launch (see debug for traceback): Caught exception when trying to load file of format [xml]: Error trying to process rule "substitution":

Unknown substitution: arg
```

**To Reproduce**
Steps to reproduce the behavior:
1. activate conda environment

```
gcervettini:~$ conda activate ros2_humble
```

2. navigate to the installation directory

```
(ros2_humble) gcervettini:~$ cd robot-code/
```

3. source the setup file

```
(ros2_humble) gcervettini:~/robot-code$ source ros2_ws/install/setup.sh
```

4. ros launch ouster_ros replay

```
(ros2_humble) gcervettini@iitbmp014lw010u:~/robot-code$ ros2 launch ouster_ros replay.launch.xml    \
    bag_file:=/rosbag2_2024_09_12-16_06_32/rosbag2_2024_09_12-16_06_32_0.db3     \
    metadata:=os-122340001722-metadata.json
```


**Info**
Here is how my [replay.launch.xml](https://github.com/ouster-lidar/ouster-ros/blob/ros2/ouster-ros/launch/replay.composite.launch.xml) looks like:  

<details>
  <summary>replay.launch.xml</summary>
  
  ```
<launch>

  <set_parameter name="use_sim_time" value="true" />

  <arg name="loop" default="false" description="request loop playback"/>
  <arg if="$(arg loop)" name="_loop" value="--loop"/>
  <arg unless="$(arg loop)" name="_loop" value=" "/>

  <remap from="/os_node/imu_packets" to="/ouster/imu_packets"/>
  <remap from="/os_node/lidar_packets" to="/ouster/lidar_packets"/>

  <arg name="ouster_ns" default="ouster"
    description="Override the default namespace of all ouster nodes"/>
  <!-- TODO: revisit the proper behaviour of allowing users override the default timestamp_mode during replay -->
  <arg name="timestamp_mode" default=""
    description="method used to timestamp measurements; possible values: {
    TIME_FROM_INTERNAL_OSC,
    TIME_FROM_SYNC_PULSE_IN,
    TIME_FROM_PTP_1588,
    TIME_FROM_ROS_TIME
    }"/>
   <arg name="ptp_utc_tai_offset" default="-37.0"
    description="UTC/TAI offset in seconds to apply when using TIME_FROM_PTP_1588"/>
  <arg name="metadata" default=""
    description="path to write metadata file when receiving sensor data"/>
  <arg name="bag_file"
    description="file name to use for the recorded bag file"/>
  <arg name="viz" default="true"
    description="whether to run a rviz"/>
  <arg name="rviz_config" default="$(find-pkg-share ouster_ros)/config/viz-reliable.rviz"
    description="optional rviz config file"/>

  <arg name="sensor_frame" default="os_sensor"
    description="sets name of choice for the sensor_frame tf frame, value can not be empty"/>
  <arg name="lidar_frame" default="os_lidar"
    description="sets name of choice for the os_lidar tf frame, value can not be empty"/>
  <arg name="imu_frame" default="os_imu"
      description="sets name of choice for the os_imu tf frame, value can not be empty"/>
  <arg name="point_cloud_frame" default=""
    description="which frame to be used when publishing PointCloud2 or LaserScan messages.
    Choose between the value of sensor_frame or lidar_frame, leaving this value empty
    would set lidar_frame to be the frame used when publishing these messages."/>

  <let name="_use_metadata_file" value="$(eval '\'$(var metadata)\' != \'\'')"/>

  <arg name="use_system_default_qos" default="true"
    description="Use the default system QoS settings"/>

  <arg name="proc_mask" default="IMG|PCL|IMU|SCAN" description="
    use any combination of the 4 flags to enable or disable specific processors"/>

  <arg name="scan_ring" default="0" description="
    use this parameter in conjunction with the SCAN flag
    and choose a value the range [0, sensor_beams_count)"/>

  <arg name="point_type" default="original" description="point type for the generated point cloud;
   available options: {
    original,
    native,
    xyz,
    xyzi,
    xyzir
    }"/>

  <group>
    <push-ros-namespace namespace="$(var ouster_ns)"/>
    <node if="$(var _use_metadata_file)" pkg="ouster_ros" exec="os_replay" name="os_replay" output="screen">
      <param name="metadata" value="$(var metadata)"/>
      <param name="use_system_default_qos" value="$(var use_system_default_qos)"/>
    </node>
    <node_container pkg="rclcpp_components" exec="component_container_mt" name="os_container" output="screen" namespace="">
      <composable_node pkg="ouster_ros" plugin="ouster_ros::OusterCloud" name="os_cloud">
        <param name="sensor_frame" value="$(var sensor_frame)"/>
        <param name="lidar_frame" value="$(var lidar_frame)"/>
        <param name="imu_frame" value="$(var imu_frame)"/>
        <param name="point_cloud_frame" value="$(var point_cloud_frame)"/>
        <param name="timestamp_mode" value="$(var timestamp_mode)"/>
        <param name="ptp_utc_tai_offset" value="$(var ptp_utc_tai_offset)"/>
        <param name="use_system_default_qos" value="$(var use_system_default_qos)"/>
        <param name="proc_mask" value="$(var proc_mask)"/>
        <param name="scan_ring" value="$(var scan_ring)"/>
        <param name="point_type" value="$(var point_type)"/>
      </composable_node>
      <composable_node pkg="ouster_ros" plugin="ouster_ros::OusterImage" name="os_image">
        <param name="use_system_default_qos" value="$(var use_system_default_qos)"/>
        <param name="proc_mask" value="$(var proc_mask)"/>
      </composable_node>
    </node_container>
  </group>

  <!-- HACK: configure and activate the replay node via a process execute since state
    transition is currently not availabe through launch.xml format -->
  <executable if="$(var _use_metadata_file)"
    cmd="$(find-exec ros2) lifecycle set /$(var ouster_ns)/os_replay configure"
    launch-prefix="bash -c 'sleep 0; $0 $@'" output="screen"/>
  <executable if="$(var _use_metadata_file)"
    cmd="$(find-exec ros2) lifecycle set /$(var ouster_ns)/os_replay activate"
    launch-prefix="bash -c 'sleep 1; $0 $@'" output="screen"/>

  <include if="$(var viz)" file="$(find-pkg-share ouster_ros)/launch/rviz.launch.xml">
    <arg name="ouster_ns" value="$(var ouster_ns)"/>
    <arg name="rviz_config" value="$(var rviz_config)"/>
  </include>

  <let name="_use_bag_file_name" value="$(eval '\'$(var bag_file)\' != \'b\'')"/>

  <executable if="$(var _use_bag_file_name)" output="screen"
    launch-prefix="bash -c 'sleep 3; $0 $@'"
    cmd="ros2 bag play $(var bag_file) --clock $(arg _loop)
      --qos-profile-overrides-path
      $(find-pkg-share ouster_ros)/config/metadata-qos-override.yaml"/>

</launch>
  ```
</details>

Looking through ROS2 Humble [documentation](https://docs.ros.org/en/humble/How-To-Guides/Migrating-from-ROS1/Migrating-Launch-Files.html), I see some details of the `xml` above that seems not to be supported after migrating from ROS1 to ROS2:  

* the [remap](https://docs.ros.org/en/humble/How-To-Guides/Migrating-from-ROS1/Migrating-Launch-Files.html#remap) tag can only be used within `node` tags.
* for the [arg](https://docs.ros.org/en/humble/How-To-Guides/Migrating-from-ROS1/Migrating-Launch-Files.html#arg) tag, `value` attribute is not allowed. The `let` tag should be used instead.

**Platform:**
- Ouster Sensor? \[OS-Dome\]
- Ouster Firmware Version? \[v3.0.1\]
- ROS version/distro? \[humble\]
- Operating System? \[Linux\]
- Machine Architecture? \[x64\]
- git commit hash? \[latest\].

--- a/ouster-ros/launch/replay.independent.launch.xml
+++ b/ouster-ros/launch/replay.independent.launch.xml
@@ -6,11 +6,6 @@
   <let if="$(var loop)" name="_loop" value="--loop"/>
   <let unless="$(var loop)" name="_loop" value=" "/>
 
-  <!--
-  <remap from="/os_node/imu_packets" to="/ouster/imu_packets"/>
-  <remap from="/os_node/lidar_packets" to="/ouster/lidar_packets"/>
-  -->
-
   <arg name="ouster_ns" default="ouster"
     description="Override the default namespace of all ouster nodes"/>
   <!-- TODO: revisit the proper behaviour of allowing users override the default timestamp_mode during replay -->
@@ -72,6 +67,8 @@
       <param name="metadata" value="$(var metadata)"/>
     </node>
     <node pkg="ouster_ros" exec="os_cloud" name="os_cloud" output="screen">
+      <remap from="/os_node/imu_packets" to="/ouster/imu_packets"/>
+      <remap from="/os_node/lidar_packets" to="/ouster/lidar_packets"/>
       <param name="sensor_frame" value="$(var sensor_frame)"/>
       <param name="lidar_frame" value="$(var lidar_frame)"/>
       <param name="imu_frame" value="$(var imu_frame)"/>
@@ -84,6 +81,7 @@
       <param name="point_type" value="$(var point_type)"/>
     </node>
     <node pkg="ouster_ros" exec="os_image" name="os_image" output="screen">
+      <remap from="/os_node/lidar_packets" to="/ouster/lidar_packets"/>
       <param name="use_system_default_qos" value="$(var use_system_default_qos)"/>
     </node>
   </group>

--- a/ouster-ros/launch/replay.independent.launch.xml
+++ b/ouster-ros/launch/replay.independent.launch.xml
@@ -3,11 +3,13 @@
   <set_parameter name="use_sim_time" value="true" />
 
   <arg name="loop" default="false" description="request loop playback"/>
-  <arg if="$(arg loop)" name="_loop" value="--loop"/>
-  <arg unless="$(arg loop)" name="_loop" value=" "/>
+  <let if="$(var loop)" name="_loop" value="--loop"/>
+  <let unless="$(var loop)" name="_loop" value=" "/>
 
+  <!--
   <remap from="/os_node/imu_packets" to="/ouster/imu_packets"/>
   <remap from="/os_node/lidar_packets" to="/ouster/lidar_packets"/>
+  -->
 
   <arg name="ouster_ns" default="ouster"
     description="Override the default namespace of all ouster nodes"/>
@@ -104,7 +106,7 @@
 
   <executable if="$(var _use_bag_file_name)" output="screen"
     launch-prefix="bash -c 'sleep 3; $0 $@'"
-    cmd="ros2 bag play $(var bag_file) --clock $(arg _loop)
+    cmd="ros2 bag play $(var bag_file) $(var _loop)
       --qos-profile-overrides-path
       $(find-pkg-share ouster_ros)/config/metadata-qos-override.yaml"/>
 

